spark-commons/src/test/scala/za/co/absa/spark/commons/errorhandling/implementations/ErrorMessageArrayTest.scala

```diff
@@ -197,6 +192,25 @@ class ErrorMessageArrayTest extends AnyFunSuite with SparkTestBase {
     val result = resultDfToResult(resultDf)
 
     assert(result == expected)
+  }
+
+  test("errorColumnAggregationType should return an ArrayType structure for column added during the aggregation") {
+    val errColName = "errCol"
+    val errorMessageArray = ErrorMessageArray(errColName)
 
+    val e1 = errorMessageArray.putErrorToColumn("Test error 1", 1, "This is a test error", Some(col1Name))
+    val errorSubmitA = ErrorMessageSubmitOnColumn("Test error 2", 2, "This is a test error", col2Name)
+    val e2 = errorMessageArray.putErrorToColumn(errorSubmitA)
+    val errorSubmitB = ErrorMessageSubmitWithoutColumn("Test error 3", 3, "This is a test error")
+    val e3 = errorMessageArray.putErrorToColumn(errorSubmitB)
+
+    val dfAfterAgg = errorMessageArray.aggregateErrorColumns(srcDf)(e1, e2, e3)
+    val addedColType = dfAfterAgg.select(errColName).schema
+
+    val results = errorMessageArray.errorColumnAggregationType
+    val actualResultsType = DataTypeImplicits.DataTypeEnhancements(addedColType).wrapStructFiledWithStructType(results, errColName)
+
+    assert(actualResultsType)
```

Comment:
This would work much better. It is more explanative and test more things.
```suggestion
    val origSchemaSize = srcDf.schema.fieldNames.length
    val dfAfterAgg = errorMessageArray.aggregateErrorColumns(srcDf)(e1, e2, e3)
    assert(dfAfterAgg.schema.fieldNames.length == origSchemaSize +1) // checkc only one field was added...
    assert(dfAfterAgg.schema.fieldNames(origSchemaSize) == errColName) // and is of correct name...
    //... and type
    val addedColType = dfAfterAgg.schema.fields(origSchemaSize).dataType

    // while using `get` in `Option` is discouraged, it's ok here, as it's expected the option to be non-empty; and if empty
    // the test will fail, which is correct
    val result = errorMessageArray.errorColumnAggregationType.get

    import DataTypeImplicits.DataTypeEnhancements
    assert(result.isEquivalentDataType(addedColType))
```